src/axom/quest/readers/PC2CReader.hpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef QUEST_PC2CREADER_HPP_
+#define QUEST_PC2CREADER_HPP_
+
+#include "axom/config.hpp"
+
+#ifndef AXOM_USE_C2C
+  #error PC2CReader should only be included when Axom is configured with C2C
+#endif
+
+#include "axom/core/Macros.hpp"
+#include "axom/quest/readers/C2CReader.hpp"  // base class
+
+#include "mpi.h"
+
+namespace axom
+{
+namespace quest
+{
+class PC2CReader : public C2CReader
+{
+public:
+  PC2CReader() = delete;
+  PC2CReader(MPI_Comm comm);
+
+  virtual ~PC2CReader() = default;
+
+  /*!
+   * \brief Reads in a C2C file to all ranks in the associated communicator
+   *
+   * \note Rank 0 reads in the C2C mesh file and broadcasts to the other ranks
+   * \return status set to zero on success; non-zero otherwise
+   */
+  virtual int read() final override;
```

Comment:
The `virtual` is redundant here, and `override` already tells the reader (and compiler) you're overriding a function. Some tools (e.g. clion) complain about this redundancy.

Reply:
Good suggestion.
I updated `PSTLReader` as well